src/common/mem.c

```diff
@@ -586,26 +585,25 @@ int
 umem_init_txd(struct umem_tx_stage_data *txd)
 {
 	D_ASSERT(txd != NULL);
-	memset(txd, 0, sizeof(*txd));
 	txd->txd_magic = UMEM_TX_DATA_MAGIC;
 
 	D_ALLOC_ARRAY(txd->txd_commit_vec, TXD_CB_NUM);
-	if (txd->txd_commit_vec == NULL)
-		goto fail;
-	txd->txd_commit_max = TXD_CB_NUM;
+	txd->txd_commit_max = txd->txd_commit_vec != NULL ? TXD_CB_NUM : 0;
```

Comment:
what is the benefit here?  If this alloc fails, why do the other two allocs ?

Reply:
There is less probability that alloc fails. I don't think it's important to spen less time if it fails overall anyway. But in good path it will be faster because of less conditions. And it's has better readability IMHO.